### PR TITLE
CF Bump to 3.6.0

### PR DESCRIPTION
**Pull request by andreas-kupries**
WIP, :construction: As it passes :cat: on vagrant (on 2nd run), now is the time to see what Jenkins thinks.

Notes
   * New releases, and ignored in this first step are

      * bosh-dns-aliases-release	0.0.3
      * log-cache-release		1.4.3

     Things seem to work without them. The `bpm-release` was needed.
     Due to most jobs in `routing-release` using bpm unconditionally. Using v1.0.0 instead of 0.12.2. IOW this one release is ahead of the deployment.

   * Still ignored releases are

      * cf-networking-release		2.14.0
      * consul-release			195
      * silk-release			2.14.0

   * The new garden-runc job

      * netplugin-server

     and the new loggregator jobs

      * reverse_log_proxy_gateway_cf_auth_proxy
      * reverse_log_proxy_gateway

     are ignored as well.

   * A few property renames can be ignored, as we are using the renamed properties at all.
     Most properties look to have sensible defaults.
    
   * Four new diego properties were switched off, i.e. their default of true got overriden to false.
      * bbs:set_kernel_parameters
      * file_server:set_kernel_parameters
      * locket:set_kernel_parameters
      * rep:set_kernel_parameters

     See the relevant description below. While we are not bosh in any way, we are containerized.
```
    	Enable tuning /proc/sys kernel parameters.
        NOTE: set this property to 'false' when deploying
        to BOSH-Lite or other containerized BOSH clouds.
```

Back to BPM. All the roles which got jobs from routing-release with BPM enabled, i.e. jobs `route-registrar`, gorouter`, `routing-api` now require `cap: ALL`, aka `privileged`, and, of course, the associated PSP for BPM to work properly. The affected roles are

   * cf-usb-group (route-registrar)
   * api-group (route-registrar)
   * blobstore (route-registrar)
   * log-api (route-registrar)
   * credhub-user (route-registrar)
   * autoscaler-postgres (route-registrar)
   * autoscaler-servicebroker (route-registrar)
   * router (gorouter)
   * routing-api (routing-api)

The bosh-dns thing we feared turned out to be a no-show. Not needed at all. Further reading during the bump made it look to be the bosh equivalent of the `kube-dns`. 

A few patches to support the DNS in the bpm containers.

### CODE REVIEW

File: container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+   capabilities:
+   - NET_BIND_SERVICE
++  unsafe:
++    unrestricted_volumes:
++    - path: /etc/hostname
++    - path: /etc/hosts
++    - path: /etc/hosts
++    - path: /etc/resolv.conf
++    - path: /etc/ssl
++    - path: /var/lib
```
Comment (mook-as) [inline container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh]: All of `/var/lib` seems a bit much; just `/var/lib/ca-certificates` should be enough.

Also, needs comments to say what these mounts are for.

(These comments apply to the other patches as well)

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 2]
```diff
@@ -388,6 +393,13 @@ instance_groups:
           internal: 24053
   - name: route_registrar
     release: routing
+  - name: bpm
+    release: bpm
+    properties:
+      bosh_containerization:
```
[the remark this answers is not in the document]
Comment (jandubois) [reply container-host-files/etc/scf/config/role-manifest.yml]: It is supposed to: https://github.com/cloudfoundry-incubator/fissile/blob/develop/model/role_run.go#L162

[the remark this answers is not in the document]
Comment (mook-as) [reply container-host-files/etc/scf/config/role-manifest.yml]: And the PSP gets merged too: https://github.com/cloudfoundry-incubator/fissile/blob/8024619e0f5d236ebf2c61f961c1362aee006efa/model/instance_groups.go#L469

LGTM on this.

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 3]
```diff
@@ -349,6 +353,7 @@ instance_groups:
   scripts:
   - scripts/forward_logfiles.sh
   - scripts/patches/fix_monit_rsyslog.sh
+  - scripts/patches/fix_rr_bpm_dns.sh
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/role-manifest.yml]: Ok.

File: container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
++++ bpm.yml.erb
+@@ -13,3 +13,11 @@ processes:
+     pre_start: /var/vcap/jobs/gorouter/bin/bpm-pre-start
+   capabilities:
+   - NET_BIND_SERVICE
++  unsafe:
++    unrestricted_volumes:
++    - path: /etc/hostname
++    - path: /etc/hosts
++    - path: /etc/hosts
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh]: Looks to be :( To fix.

File: container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh (hunk 1 above)
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh]: +1 on comments. Restricting the mount will require another round of testing

File: container-host-files/etc/scf/config/role-manifest.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/role-manifest.yml]: Fixed by Jan with commit 5e020dbe79029da6d3c00d7bd02f23f6c23459e7

File: container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh]: Fixed by Jan with commit 5e020dbe79029da6d3c00d7bd02f23f6c23459e7

File: container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh (hunk 1 above)
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh]: Mount change fixed by Jan with commit 5e020dbe79029da6d3c00d7bd02f23f6c23459e7.
Comments to come.

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 5]
```diff
@@ -1,53 +1,57 @@
...
+  sha1: "4236495d4a047fb9396996aba3daea9e0cb3f1bf"
 - name: cf-mysql
   url: https://bosh.io/d/github.com/cloudfoundry/cf-mysql-release?v=36.15.0
   version: 36.15.0
   sha1: 0764d9d6aae7cefd10019437ed83e7715e614633
-- name: cf-smoke-tests
-  url: https://bosh.io/d/github.com/cloudfoundry/cf-smoke-tests-release?v=40.0.5
-  version: 40.0.5
-  sha1: 6f6081d24490a82c5b9fea73283e0cfe21cbf207
+- name: "cf-smoke-tests"
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/role-manifest.yml]: Fixes to come.

[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/role-manifest.yml]: See https://github.com/SUSE/scf/pull/1956

File: container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh (hunk 1 above)
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/scripts/patches/fix_gorouter_bpm_dns.sh]: See https://github.com/SUSE/scf/pull/1956

**Pull request merged**